pymeasure/adapters/protocol.py

```
@@ -123,16 +127,17 @@ def _write_bytes(self, content, **kwargs):
         try:
             p_write, p_read = self.comm_pairs[self._index]
         except IndexError:
-            raise ValueError(f"No communication pair left to write {content}.")
-        if self._write_buffer == to_bytes(p_write):
+            raise ValueError(f"No communication pair left to write '{content!r}'.")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)